### PR TITLE
Handle container types within container types

### PR DESCRIPTION
**Pull request by roshan**
I have a `Set<List<Enum>>` in a Thrift object, and the current `master` fails to generate for that. This is intended to fix that.
I also added in `Map` functionality while I was there but not any shrinking.

### CODE REVIEW

**Comment by roshan** [comment]
@pwestling would you mind taking a look?

File: generative_core/src/main/java/com/liveramp/generative/MapOf.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class MapOf<K, V> implements Arbitrary<Map<K, V>> {
+
+  private final Arbitrary<K> keyGen;
+  private final Arbitrary<V> valueGen;
+  private final int numKvPairs;
+
+  public MapOf(Arbitrary<K> keyGen, Arbitrary<V> valueGen, int numKvPairs) {
```
Comment (pwestling) [inline generative_core/src/main/java/com/liveramp/generative/MapOf.java]: We should change the "length" argument here to itself be an arbitrary. We can keep this constructor and have it make a `new Fixed` for the length

File: generative_core/src/main/java/com/liveramp/generative/MapOf.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    this.valueGen = valueGen;
+    this.numKvPairs = numKvPairs;
+  }
+
+  @Override
+  public Map<K, V> get(Random r) {
+    return IntStream.range(0, numKvPairs)
+        .boxed()
+        .collect(Collectors.toMap(
+            i -> keyGen.get(r),
```
Comment (pwestling) [inline generative_core/src/main/java/com/liveramp/generative/MapOf.java]: This is a little weird in that if you provide a fixed size for your map, the generated map won't actually always have that many key value pairs in the case that keys collide. Possibly this is fine, but my gut would be to generate a Set of the desired value count, then assign random values. I'm realizing SetOf may also have this flaw.

Comment (pwestling) [reply generative_core/src/main/java/com/liveramp/generative/MapOf.java]: SetOf did have this flaw. Fix here: https://github.com/LiveRamp/generative/pull/3

**Comment by pwestling** [comment]
I'm on vacation starting today. I trust you on this one. Merge at your
discretion.

**Comment by roshan** [comment]
Muchos gracias, autosquashing and rebased to `master`.

**Pull request merged**